File: MobileClient/Class/Product.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MobileClient.Class
{
    public class Product
    {
        [JsonIgnore]
        internal static ObservableCollection<Product> Products = new ObservableCollection<Product>();
        public string Name { get; set; } = string.Empty;
        public string Barcode { get; set; } = string.Empty;
        public string Shop { get; set; } = string.Empty;
        public double Price { get; set; } = 0;
        [JsonIgnore]
        public string ShowNameCart { get { return $"{Name} w Sklepie {Shop}."; } }
        [JsonIgnore]
        public string ShowName { get { return $"{Name} - w Sklepie {Shop} za {Price} zł"; } }
        [JsonIgnore]
        public string ShowNameLong { get { return $"{Name} - w Sklepie {Shop} za {Price} zł w ilości {Qty}"; } }
        public double Qty { get; set; } = 0;

        [JsonIgnore]
        public bool IsPurchased { get; set; } = false;

        public Product() { }

        public Product(string name, string barcode, string shop, double price, int qty)
        {
            Name = name;
            Barcode = barcode;
            Shop = shop;
            Price = price;
            Qty = qty;
        }

        public string ToJson()
        {
            return JsonConvert.SerializeObject(this);
        }

        public static Product FromJson(string json)
        {
            return JsonConvert.DeserializeObject<Product>(json);
        }

        public static ObservableCollection<Product> FromJsonList(string json)
        {
            return JsonConvert.DeserializeObject<ObservableCollection<Product>>(json);
        }

        public static string ListToJson(ObservableCollection<Product> list)
        {
            return JsonConvert.SerializeObject(list);
        }
    }
}

File: WorkerService/Product.fs
namespace WorkerService

open Newtonsoft.Json
open System.Collections.ObjectModel

type Product(nameIn: string, barcodeIn: string, priceIn: double, shopIn: string, qtyIn: double) =
    let mutable name: string = ""
    let mutable barcode: string = ""
    let mutable price: double = -1
    let mutable shop: string = ""
    let mutable qty: double = -1
    do
        name <- nameIn
        barcode <- barcodeIn
        price <- priceIn
        shop <- shopIn
        qty <- qtyIn

    member this.Name with get() = name and set(value) = name <-value
    member this.Barcode with get() = barcode and set(value) = barcode <-value
    member this.Shop with get() = shop and set(value) = shop <-value
    member this.Price with get() = price and set(value) = price <-value
    member this.Qty with get() = qty and set(value) = qty <-value

    member this.ToJson() =
        JsonConvert.SerializeObject(this)

    static member FormJson(json: string) : Product =
        JsonConvert.DeserializeObject<Product>(json)

    static member FormJsonProducts(json: string) : ObservableCollection<Product> =
        JsonConvert.DeserializeObject<ObservableCollection<Product>>(json)

    static member SerializeToJson(list: ObservableCollection<Product>) =
        JsonConvert.SerializeObject(list)
        
